frontend/public/components/chargeback.tsx

```diff
@@ -282,31 +272,40 @@ class ReportData extends React.Component<ReportDataProps, ReportDataState> {
         "pod_request_memory_byte_seconds": 23385341952000
       },
       All fields but start/end are optional. */
-    _.each(data, row => {
-      const key = row[reduceBy];
-      // key == 'namespace', 'pod', 'node' (whatever we're aggregating by)
-      if (!reducedData[key]) {
-        reducedData[key] = {};
-      }
-      _.each(row, (v, k) => {
-        if (!isFinite(v as (any))) {
-          return;
-        }
-        // k == 'pod_memory_usage_percent', 'cost', etc (all columns in table except 1st)
-        if (!reducedData[key][k]) {
-          reducedData[key][k] = 0;
-        }
-        reducedData[key][k] += v;
-        if (!totals[k]) {
-          totals[k] = 0;
-        }
-        totals[k] += v;
-        if (reducedData[key][k] > (maxValues[k] || 0)) {
-          maxValues[k] = reducedData[key][k];
+
+    const canReduceData:boolean = !_.isEmpty(data) && _.get(data, [0, reduceBy]) !== undefined;
+    if (canReduceData) {
+      _.each(data, row => {
+        const key = row[reduceBy];
+        // key == 'namespace', 'pod', 'node' (whatever we're aggregating by)
+        if (!reducedData[key]) {
+          reducedData[key] = {};
         }
+        _.each(row, (v, k) => {
+          if (!isFinite(v as (any))) {
+            return;
+          }
+          // k == 'pod_memory_usage_percent', 'cost', etc (all columns in table except 1st)
+          if (!reducedData[key][k]) {
+            reducedData[key][k] = 0;
+          }
+          reducedData[key][k] += v;
+          if (!totals[k]) {
+            totals[k] = 0;
+          }
+          totals[k] += v;
+          if (reducedData[key][k] > (maxValues[k] || 0)) {
+            maxValues[k] = reducedData[key][k];
+          }
+        });
+      });
+    }
+    const rows = canReduceData ? _.orderBy(_.map<any>(reducedData, (o, key) => ({[reduceBy]: key, ...o})), sortBy, orderBy)
```

Comment:
Can you describe what this is doing? This is unclear to me.

Reply:
We are either reducing the data (aggregating totals and max value for 'bars' in cell), or return the raw data (minus the blacked out columns/fields).